woocommerce/woocommerce
# Refactor Country Specs to use in operator

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




Closes #51129.

This PR partially addresses the issue reported in https://github.com/woocommerce/woocommerce/issues/50714#issuecomment-2313703493. Country spec checks are implemented with long lists of operators.

This PR refactors the country spec checks to use a `in` operator instead to improve the performance of the checks.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with this branch
2. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
3. Turn off `Display suggestions within WooCommerce`
4. Go to Core Profiler and choose `Afghanistan` as the country
5. Continue to Plugin installation step
6. Confirm WooPayments, TikTok, and WooCommerce Shipping & Tax are NOT suggested
7. Repeat the test with `Australia` and confirm that WooPayments, TikTok, and WooCommerce Shipping & Tax are suggested
8. Skip core profiler
9. Go to `WooCommerce > Home``
10. For AU, you should see the shipping task in the task list, and going to it should show you `easyship` as a double partner alongside `shipstation`

![Screenshot 2024-09-05 at 14 29 09](https://github.com/user-attachments/assets/4170aacf-9221-4c82-93b6-956e6345049f)


![Screenshot 2024-09-05 at 14 00 21](https://github.com/user-attachments/assets/c0958596-e366-4695-9ef3-362a662f5853)

![Screenshot 2024-09-05 at 14 13 05](https://github.com/user-attachments/assets/24cc7fb7-8f1b-4578-bb3b-f9564a6b06a7)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
